UofS-Pulse-Binfo/uofspb_germplasm
# Updated Germplasm Collection create and view functionality

This PR creates functionality to create and view germplasm collection.


## Metadata

This fixes the issue https://github.com/UofS-Pulse-Binfo/uofspb_germplasm/issues/13




Documentation:

## Description


Created a custom Tripal Field where the formatter is configured to view collection and widget is setup to create/insert germplasm to a collection.

## Dependencies



## Testing?




1. Clone uofspb_germplasm and switch to branch add-stock-to-collection.
2. Start a KP clone and point the volume to the module in Step 1 (see KP Clone repository). 
3. Navigate to Structure/Tripal Content Types and select managed fields for Germplasm Collection.

4. If it was installed previously, remove the field Germplasm List co_0010__germplasm by clicking delete link under the OPERATIONS column in Germplasm Collection field manager page.

![image](https://user-images.githubusercontent.com/15472253/202260515-a51e44bd-583f-4f1c-825f-a6bd4725e645.png)

5. Re-install the Germplasm Collection Field. This ensures that field variables are properly initiated and additional functionality is recognized by Drupal.

![image](https://user-images.githubusercontent.com/15472253/202262771-4ba592ce-81f5-428e-b8a4-f4a4c434f2fa.png)

Type: Germplasm List or Germplasm Collection in the Add new field and select Germplasm List in the field type selectbox. Widget for this field will default to Germplasm List.

Click ![image](https://user-images.githubusercontent.com/15472253/202263282-d63cb6a4-8df2-484f-a624-9e660855b68c.png) (Which is located at the bottom of the page)

After clicking Save button, you will be redirected to Germplasm Collection Field Settings page. Accept all default values for this field and scroll down to bottom of the page and click ![image](https://user-images.githubusercontent.com/15472253/202268198-f809b884-4dc1-4943-a9ff-02740cce411d.png)

6. Once saved, navigate back to field manager page and click on Manage Display tab and inspect the newly created field if it ended up under Disabled fields. 
![image](https://user-images.githubusercontent.com/15472253/202270931-15835ed1-c3a0-4195-bddb-9fec59299389.png)

If this is the case, click and drag crosshair icon inline with the field name and move it out of the disabled section. Click Save button once more.
![image](https://user-images.githubusercontent.com/15472253/202263661-44b8cf92-0fff-4d17-beaf-0b7df9d77665.png)

7. Clear the cache. DO NOT SKIP THIS STEP


**TO VIEW/CREATE A COLLECTION:**
To add collection, navigate to Content/Tripal Content/Add Tripal Content and select Germplasm Collection in the list.
To view collection, navigate to Content/Tripal Content and filter by Germplasm Collection 

![image](https://user-images.githubusercontent.com/15472253/202270153-5968ff72-7fd6-4e76-8909-4e52103ea2d7.png)


NOTE: peculiarity seen when adding the field through Check for new fields link. It will set the field type to Chado Property and would set the formatter to Chado Property as a result, you could create a collection but unable to view it.
